File: ccc/clion/algo/birthday_chocolates.cpp
#include <iostream>
#include <vector>
#include <time.h>
using namespace std;


int birthday(vector<int> s, int d, int m) {
    int count = 0;
    for (int i = 0; i < s.size(); i++) {
        int sum = 0;
        for (int j = i; j < i + m; j++) {
            sum += s[j]; 
        }
        if (sum == d) count++;
    }
    return count;
} 

int main() {
    clock_t start, end;
    start = clock();
    int a;
    cin >> a;
    vector<int> s(a);
    for (int i = 0; i < a; i++) {
        int a;
        cin >> a;
        s[i] = a;
    }
    int d, m;
   dcin >> d >> m;
    cout << birthday(s, d, m) << endl;
    end = clock();
    double cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
    cout << "\n" << endl;
    cout << cpu_time_used << endl;
    
    return 0;
}

File: ccc/clion/algo/birthday_chocolates.py
from time import time
def birthday(s, d, m):
	count = 0
	for i in range(len(s)):
		sum = 0
		for j in range(i, i + m):
			if j < len(s):
				sum += s[j]
		if sum == d:
			count += 1
	return count

start = time()
s = [1,2,2,2,3,4,5,5,5,5,5,8,2,2,7,4,2,3,1,2,2,2,3,4,5,5,5,5,5,8,2,2,7,4,2,3,3,1]
d = 4
m = 1
print(birthday(s,d,m))
print(time() - start)

